internal/datanode/services.go

```diff
@@ -776,6 +777,26 @@ func saveSegmentFunc(node *DataNode, req *datapb.ImportTaskRequest, res *rootcoo
 		log.Info("adding segment to the correct DataNode flow graph and saving binlog paths", logFields...)
 
 		err := retry.Do(context.Background(), func() error {
+			clusteringInfo, err := clustering.ClusteringInfoFromKV(req.GetImportTask().GetInfos())
+			if err != nil {
+				log.Warn("failed to parse clustering info from import request", zap.Error(err))
+				return fmt.Errorf(err.Error())
+			}
+			if clusteringInfo != nil && clusteringInfo.Id == 0 {
```

Comment:
should be groupId? if id is not set, we should not alloc one, we should fail

Reply:
Here is clusteringInfo.id, allocated by milvus.  `groupId` is passed by user. I haven't implement `ClusteringGroup` logic yet, so `groupId` is not used now.